test/controllers/timezone_controller_test.exs

```diff
@@ -9,9 +9,33 @@ defmodule Ask.TimezoneControllerTest do
     {:ok, conn: conn, user: user}
   end
 
-  test "returns Timex timezones", %{conn: conn} do
-    conn = get conn, timezone_path(conn, :timezones)
-    assert json_response(conn, 200)["timezones"] == Timex.timezones
-  end
+  describe "the timezones map" do
+    test "should link each deprecated timezone with its canonical timezone", %{conn: conn} do
+      conn = get conn, timezone_path(conn, :timezones)
+      timezones = json_response(conn, 200)["timezones"]
+
+      Tzdata.zone_alias_list()
+      |> Enum.each(fn deprecated_tz ->
+        assert timezones[deprecated_tz] == Tzdata.links()[deprecated_tz]
+      end)
+    end
+
+    test "should link each canonical timezone with itself", %{conn: conn} do
```

Comment:
I would add a test with a canonical timezone hardcoded. Something like "should link `America/New_York` with itself"

So we cover a specific use case and avoid making all tests depending on the same libraries used in the code we're testing.

Reply:
The thing about doing this (or what you suggested in the comment above) is that perhaps in the future "America/NewYork" is no longer a canonical timezone. That is why I didn't test a specific timezone but test it against the Tzdata library that updates all the timezones data from the IANA everyday (or something like that)